### PR TITLE
Small bug fixes per user complaints and minor refactoring

### PR DESCRIPTION
**Pull request by atallahhezbor**
* Fixes an issue where sign-up form errors were not being displayed. The javascript existed to display them but the html elements were missing (likely a result of the redesign)

* The javascript to load course reviews was running on every page of the courses controller (ie each step of the browse workflow). It's only needed when you bottom-out at the section, so added code to look for the grade wheel before continuing.

### CODE REVIEW

**Comment by kienan** [comment]
Can you update the PR body with details?

File: app/assets/stylesheets/welcome.scss [hunk 1]
```diff
@@ -34,4 +34,9 @@
   margin-top: 5px;
   color: white;
   padding-top: 20px;
+}
+
+.form-error {
+  display: none;
+  color: black;
 }
```
Comment (kienan) [inline app/assets/stylesheets/welcome.scss]: Should probably just add an EOL while we're at it.

File: app/views/welcome/index.html.slim [hunk 2]
```diff
@@ -48,17 +48,27 @@
             .row.form-row
               .col-xs-6
                 = f.text_field :first_name, placeholder: "First Name", class: "form-control"
+              .col-xs-6.form-error.first-name-error
```
Comment (kienan) [inline app/views/welcome/index.html.slim]: This should probably be .col-xs-12 to make sure it shows up below?

**Comment by kienan** [comment]
LGTM pending the first-name-error change!

File: app/views/welcome/index.html.slim (hunk 2 above)
Comment (atallahhezbor) [inline app/views/welcome/index.html.slim]: @kienan col-xs-12 knocked the last name field down more than i would like. Pushing a better solution soon (moving the errors to their own row)

**Comment by kienan** [comment]
@atallahhezbor Submitting seems to clear these errors -- can we retain them somehow. 

Also, the first name error message doesn't seem to appear anymore.

**Comment by atallahhezbor** [comment]
@kienan Yeah we can use form errors and display flash alerts

Though we'd lose history, especially since this isn't a specific feature branch

File: app/controllers/registrations_controller.rb [hunk 3]
```diff
@@ -22,6 +22,7 @@ def create
       end
     else
       clean_up_passwords @user
+      flash[:notice] = flash[:notice].to_a.concat(resource.errors.full_messages).join(", ")
```
Comment (kienan) [inline app/controllers/registrations_controller.rb]: Can you check edge cases for this? I get an error sometimes when I hit submit multiple times in a row.

**Comment by kienan** [comment]
LGTM after you rebase!

**Pull request merged**